core/src/main/java/org/apache/accumulo/core/metadata/schema/AmpleImpl.java

```diff
@@ -42,4 +43,9 @@ public TabletMetadata readTablet(KeyExtent extent, ColumnType... colsToFetch) {
       return Iterables.getOnlyElement(tablets);
     }
   }
+
+  public TableOptions readTablets() {
+    TableOptions builder = TabletsMetadata.builder();
```

Comment:
The main difference with moving this Ampl is that the client can be supplied to the builder implementation at the beginning and the final method in the builder can go from `build(client)` to `build()` with no arguments.

I would try experimenting with making the following change and removing the client parameter from the build method.

```suggestion
    TableOptions builder = TabletsMetadata.builder(client);
```

Reply:
@keith-turner I've tried to see if this would work without modifying TabletsMetadata. but I have not come up with anything yet. The builder() function returns a new Builder(), but Builder() is private. Builder()'s build() method is the only public build method. This is also the only way so far to access forTable() and its associated implementation. I will keep looking at it, keeping in mind our goal to keep the implementation in TabletsMetadata.

Reply:
Its ok to modify TabletsMetadata.

Reply:
@keith-turner I am pushing another Work-In-Progress PR. I changed the access modifiers in TabletsMetadata, which might not be a good/acceptable idea. However, this allowed builder to be able to be implmented in Ample. This also allows the readTablets() method in Ample to access forTable method in TabletsMetadata.

Reply:
@keith-turner This solution would also involved modifying classes that use TabletsMetadata.build(), to use the new one off of Ample. For the time being, TabletsMetadata still has its old build() method. to keep those classes from breaking.